### DISCUSSION TITLE
Can I adjust p value in correlation-based network analysis?

### DISCUSSION THREAD

**Discussion by rainboy352900** [ChiLiubio/microeco#114] [Q&A] [answered]
Hi Liu,

I'm wondering whether I can adjust p value in  correlation-based network analysis? Such as using false discovery rate method. I did not find the related function in trans_network$new.

Thanks.

**Comment by ChiLiubio** [comment, marked as answer]
Hi @rainboy352900 
It is in the cal_network function COR_p_adjust  parameter.

Chi

**Reply by rainboy352900** [reply]
Thanks Chi.

However, I came across an error when running "get_node_table" function using my data (attached). The example data seems ok. Cound you please suggest how to deal with it?

![image](https://user-images.githubusercontent.com/99855027/172041487-730f9cff-650e-4a26-9232-9f6898471175.png)

code:
data<- read.csv(file="otu.csv",header=T, row.names=1)
dataset <- microtable$new(otu_table = data)
t1 <- trans_network$new(dataset = dataset, cal_cor = "base", taxa_level = "OTU", filter_thres = 0, cor_method = "spearman")
t1$cal_network(COR_p_thres = 0.05,COR_p_adjust = "fdr", COR_cut = 0.8)
t1$cal_module(method = "cluster_fast_greedy")
t1$cal_network_attr()
t1$res_network_attr
t1$get_node_table(node_roles = T)

data:
[otu.csv](https://github.com/ChiLiubio/microeco/files/8839536/otu.csv)

**Reply by ChiLiubio** [reply]
Hi @rainboy352900 
Sorry. This is a bug. The get_node_table function automatically merge the tax_table in dataset to the node property table. As there is no tax_table in your dataset, the error is there. I ignored this case without the tax_table. I updated the package in the github to fix it. But there are other changes in the trans_network class in the coming v0.11.0. So I donot recommend you install the developing version in github. Here is a temporary solution to fix this by adding a simple tax_table:

```
data<- read.csv(file="otu.csv",header=T, row.names=1)
dataset <- microtable$new(otu_table = data)
# add a tax_table
tax_table <- data.frame(OTU=rownames(data))
rownames(tax_table) <- rownames(data)
dataset$tax_table <- tax_table

t1 <- trans_network$new(dataset = dataset, cal_cor = "base", taxa_level = "OTU", filter_thres = 0, cor_method = "spearman")
t1$cal_network(COR_p_thres = 0.05, COR_p_adjust = "fdr", COR_cut = 0.8)
t1$cal_module(method = "cluster_fast_greedy")
t1$cal_network_attr()
t1$res_network_attr
t1$get_node_table(node_roles = T)
```

Best
Chi

**Reply by rainboy352900** [reply]
Thanks Chi. Now it works well.

**Reply by rainboy352900** [reply]
Hi Chi,

Sorry to bother you again.

In your FEMS Microbiology Ecology paper, Figure 5b was generated by "cal_sum_links", but I do not find it in v0.10.1. Is there any alternative function available?

Thanks.

**Reply by ChiLiubio** [reply]
Hi @rainboy352900 
Please see the mecodev package part of the tutorial. I moved this part to trans_netchord class in mecodev package since I think this part is not so important as other functions. But now I find many are interested in this chord plot. So I decide to move back this to microeco package in the coming v0.11.0 :)

Chi

**Reply by rainboy352900** [reply]
Thanks Chi.

I came across an error when running "trans_netchord$new" function in "mecodev" package. The example data seems ok. Cound you please suggest how to deal with it? Thanks.

![image](https://user-images.githubusercontent.com/99855027/172604371-d501a872-77ea-4b52-8098-db8e40c60139.png)

library(microeco)
library(mecodev)
data<- read.csv(file="otu.csv",header=T, row.names=1)
taxonomy_table_16S<- read.csv(file="taxonomy.csv",header=T, row.names=1)
dataset <- microtable$new(otu_table = data, tax_table = taxonomy_table_16S)
t1 <- trans_network$new(dataset = dataset, cal_cor = "base", taxa_level = "OTU", filter_thres = 0, cor_method = "spearman")
t1$cal_network(COR_p_thres = 0.05, COR_p_adjust = "fdr", COR_optimization = TRUE)
test1 <- trans_netchord$new(dataset = t1, taxa_level = "Phylum")

[otu.csv](https://github.com/ChiLiubio/microeco/files/8861037/otu.csv)
[taxonomy.csv](https://github.com/ChiLiubio/microeco/files/8861039/taxonomy.csv)

**Reply by ChiLiubio** [reply]
Hi @rainboy352900 

I test all the steps and do not get the error. The microeco version I use is v0.10.1. Please check your version and try again.

**Reply by rainboy352900** [reply]
Thanks Chi. I'll try it again.
